Source/WebCore/platform/text/TextFlags.cpp

```diff
@@ -60,48 +58,35 @@ WTF::TextStream& operator<<(TextStream& ts, Kerning kerning)
     return ts;
 }
 
-template <typename T> using BinOp = std::function<T(T, T)>;
-
-template <typename T>
-T reduce(const std::vector<T>& vec, const BinOp<T>& binOp, std::optional<T> initial = { })
-{
-    if (vec.empty())
-        return initial ? *initial : T { };
-
-    auto accumulator = initial ? *initial : vec[0];
-    auto start = initial ? 0 : 1;
-    for (size_t i = start ; i < vec.size() ; i++)
-        accumulator = binOp(accumulator, vec[i]);
-
-    return accumulator;
-}
-
 WTF::TextStream& operator<<(TextStream& ts, FontVariantAlternates alternates)
 {
     if (alternates.isNormal())
         ts << "normal";
     else {
+        StringBuilder builder;
+        auto add = [&builder](auto&& ...args) { 
+            if (!builder.isEmpty())
+                builder.append(" ");
+
+            builder.append(args...);
```

Comment:
I suspect we need  to use `std::forward` here to avoid copying arguments? I’d like to confirm we are not creating and destroying copies of `String` objects.